city_metrix/layers/layer.py

```diff
@@ -325,6 +325,18 @@ def get_stats_funcs(stats_func):
         return [stats_func]
 
 
+def set_resampling_method(image: ee.Image, resampling_method: str):
+    valid_raster_resampling_methods = ['bilinear', 'bicubic']
```

Comment:
Should you restrict the resampling methods on the dataset level instead of in this function? I think some datasets wouldn't want these resampling method (e.g. if they're not continuous).

Reply:
Wherever possible, I am striving to implement as much code as possible in layer functions so that the logic does not need to be separately implemented in multiple sub-classes.

As you mentioned, this function can only be used for continuous raster. I am renaming the function and adding documentation to further clarify the point.